### PR TITLE
Docs: Link to search with query rules page from query rules DSL

### PR DESCRIPTION
**Pull request by kderusso**
Minor docs polishing.

### CODE REVIEW

File: docs/reference/query-dsl/rule-query.asciidoc [hunk 1]
```diff
@@ -26,8 +33,8 @@ PUT _query_rules/my-ruleset
       "criteria": [
         {
           "type": "exact",
-          "metadata": "query_string",
-          "values": ["pugs"]
+          "metadata": "user_query",
```
Comment (carlosdelest) [inline docs/reference/query-dsl/rule-query.asciidoc:36]: Nice clarification 👍

**Review by carlosdelest — APPROVED** [review]

File: docs/reference/query-dsl/rule-query.asciidoc [hunk 2]
```diff
@@ -7,10 +7,17 @@
...
 
-Applies rules configured in the <<query-rules-apis, Query Rules API>> to the query.
+Applies <<query-rules-apis,query rules>> to the query before returning results.
 This feature is used to promote documents in the manner of a <<query-dsl-pinned-query>> based on matching defined rules.
 If no matching query rules are defined, the "organic" matches for the query are returned.
 
+[NOTE]
+====
+In order to use the rule query, you must have a query ruleset defined with rules to query.
+You can use the <<query-rules-apis, query rules management APIs>> to create and manage query rules.
```
Comment (leemthompo) [inline docs/reference/query-dsl/rule-query.asciidoc:17]:
```suggestion
To use the rule query, you first need a defined set of query rules.
Use the <<query-rules-apis, query rules management APIs>> to create and manage query rules.
```

**Review by leemthompo — APPROVED** [review]
Made one totally optional minor edit for clarity. I found line 16 to be a little confusing, so tried to simplify it and use parallel wording to the following sentence. :)

File: docs/reference/query-dsl/rule-query.asciidoc (hunk 2 above)
Comment (kderusso) [reply docs/reference/query-dsl/rule-query.asciidoc:17]: I like that, thank you!

**Pull request merged**